### PR TITLE
DataNode: After preflight, automatically provision new DataNodes

### PR DESCRIPTION
**Pull request by janheise**
## Description

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java [hunk 1]
```diff
@@ -17,13 +17,15 @@
...
+import org.graylog2.cluster.preflight.DataNodeProvisioningConfig;
 
 import java.security.cert.X509Certificate;
 import java.util.List;
 
 public interface CertRenewalService {
     void checkCertificatesForRenewal();
     void initiateRenewalForNode(String nodeId);
-    List<Pair<Node, X509Certificate>> findNodesAndCertificates();
+    List<Triple<Node, DataNodeProvisioningConfig, X509Certificate>> findNodes();
```
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java:30]: When I see a Triple I am thinking if this is the right moment to introduce a class/record with better interfaces for this data structure. Should I try to adapt the code slightly?

File: graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/web/resources/CertificateRenewalResource.java [hunk 2]
```diff
@@ -59,22 +59,24 @@ public CertificateRenewalResource(final TransportAddressSanitizer transportAddre
         this.nodeService = nodeService;
     }
 
-    record DataNode(String nodeId, Node.Type type, String transportAddress, DataNodeProvisioningConfig.State status, String errorMsg, String hostname, String shortNodeId, String cert) {}
+    record DataNode(String nodeId, Node.Type type, String transportAddress, DataNodeProvisioningConfig.State status, String errorMsg, String hostname, String shortNodeId, String certValidUntil) {}
```
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/web/resources/CertificateRenewalResource.java:62]: should the validity have a better type than String?

**Review by todvora — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java (hunk 1 above)
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java:30]: sure, go ahead

Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java:30]: maybe pull down the `DataNode` record from `CertificateRenewalResource`?

Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java]: changed it

**Review by dennisoelkers — APPROVED** [review]
Very good! Thank you!

**Pull request merged**